picard/ui/scripteditor.py

```diff
@@ -0,0 +1,547 @@
+    """File Naming Script Editor Page
+    """
+    NAME = "scripteditor"
+    TITLE = N_("File naming script editor")
+    PARENT = None
+    HELP_URL = '/config/options_filerenaming.html'
+    STYLESHEET_ERROR = OptionsPage.STYLESHEET_ERROR
+
+    options = [
+        TextOption(
+            "setting",
+            "file_naming_format",
+            DEFAULT_FILE_NAMING_FORMAT,
+        ),
+    ]
+
+    signal_save = QtCore.pyqtSignal()
+    signal_update = QtCore.pyqtSignal()
+
+    default_script_directory = os.path.normpath(QtCore.QStandardPaths.writableLocation(QtCore.QStandardPaths.DocumentsLocation))
+    default_script_filename = "picard_naming_script.pts"
+
+    def __init__(self, parent=None, examples=None):
+        """Stand-alone file naming script editor.
+
+        Args:
+            parent (QMainWindow or OptionsPage, optional): Parent object. Defaults to None.
+            examples (ScriptEditorExamples, required): Object containing examples to display. Defaults to None.
+        """
+        super().__init__(parent)
+        self.examples = examples
+        # TODO: Make this work properly so that it can be accessed from both the main window and the options window.
+        # self.setWindowFlags(QtCore.Qt.Window)
+        self.setWindowModality(QtCore.Qt.WindowModal)
+        self.setWindowTitle(self.TITLE)
+        self.displaying = False
+        self.ui = Ui_ScriptEditor()
+        self.ui.setupUi(self)
+
+        self.installEventFilter(self)
+
+        self.ui.file_naming_editor_save.clicked.connect(self.save_script)
+
+        self.ui.file_naming_format.setEnabled(True)
+
+        # Add scripting documentation to parent frame.
+        doc_widget = ScriptingDocumentationWidget(self)
+        self.ui.documentation_frame_layout.addWidget(doc_widget)
+        self.ui.show_documentation.stateChanged.connect(self.toggle_documentation)
+
+        self.ui.file_naming_format.textChanged.connect(self.check_formats)
+        self.ui.file_naming_word_wrap.stateChanged.connect(self.toggle_wordwrap)
+        self.ui.import_script.clicked.connect(self.import_script)
+        self.ui.export_script.clicked.connect(self.export_script)
+
+        self.ui.example_filename_sample_files_button.clicked.connect(self.update_example_files)
+        self._sampled_example_files = []
+
+        self.ui.example_filename_after.itemSelectionChanged.connect(self.match_before_to_after)
+        self.ui.example_filename_before.itemSelectionChanged.connect(self.match_after_to_before)
+
+        self.ui.preset_naming_scripts.clear()
+        for item in PRESET_SCRIPTS:
+            self.ui.preset_naming_scripts.addItem(item["title"])
+        self.ui.preset_naming_scripts.setCurrentIndex(0)
+        self.ui.preset_naming_scripts.currentIndexChanged.connect(self.select_script)
+
+        self.synchronize_vertical_scrollbars((self.ui.example_filename_before, self.ui.example_filename_after))
+
+        self.wordwrap = QtWidgets.QTextEdit.NoWrap
```

Comment:
Is this really needed?

Reply:
Nope.  Left over from an earlier version of the code.  Removed.